src/irr/asset/CGLIWriter.cpp

```diff
@@ -0,0 +1,327 @@
+
+			case EF_R16_SNORM: return getTranslatedFinalFormat(FORMAT_R16_SNORM_PACK16);		//GL_R16_SNORM
+			case EF_R16G16_SNORM: return getTranslatedFinalFormat(FORMAT_RG16_SNORM_PACK16);		//GL_RG16_SNORM
+			case EF_R16G16B16_SNORM: return getTranslatedFinalFormat(FORMAT_RGB16_SNORM_PACK16);		//GL_RGB16_SNORM
+			case EF_R16G16B16A16_SNORM: return getTranslatedFinalFormat(FORMAT_RGBA16_SNORM_PACK16);		//GL_RGBA16_SNORM
+
+			// unsigned integer formats
+			case EF_R8_UINT: return getTranslatedFinalFormat(FORMAT_R8_UINT_PACK8);				//GL_R8UI
+			case EF_R8G8_UINT: return getTranslatedFinalFormat(FORMAT_RG8_UINT_PACK8);				//GL_RG8UI
+			case EF_R8G8B8_UINT: return getTranslatedFinalFormat(FORMAT_RGB8_UINT_PACK8);			//GL_RGB8UI
+			case EF_R8G8B8A8_UINT: return getTranslatedFinalFormat(FORMAT_RGBA8_UINT_PACK8);			//GL_RGBA8UI
+
+			case EF_R16_UINT: return getTranslatedFinalFormat(FORMAT_R16_UINT_PACK16);				//GL_R16UI
+			case EF_R16G16_UINT: return getTranslatedFinalFormat(FORMAT_RG16_UINT_PACK16);			//GL_RG16UI
+			case EF_R16G16B16_UINT: return getTranslatedFinalFormat(FORMAT_RGB16_UINT_PACK16);			//GL_RGB16UI
+			case EF_R16G16B16A16_UINT: return getTranslatedFinalFormat(FORMAT_RGBA16_UINT_PACK16);			//GL_RGBA16UI
+
+			case EF_R32_UINT: return getTranslatedFinalFormat(FORMAT_R32_UINT_PACK32);				//GL_R32UI
+			case EF_R32G32_UINT: return getTranslatedFinalFormat(FORMAT_RG32_UINT_PACK32);			//GL_RG32UI
+			case EF_R32G32B32_UINT: return getTranslatedFinalFormat(FORMAT_RGB32_UINT_PACK32);			//GL_RGB32UI
+			case EF_R32G32B32A32_UINT: return getTranslatedFinalFormat(FORMAT_RGBA32_UINT_PACK32);			//GL_RGBA32UI
+
+			case EF_A2R10G10B10_UINT_PACK32: return getTranslatedFinalFormat(FORMAT_RGB10A2_UINT_PACK32);			//GL_RGB10_A2UI
+			case EF_A2R10G10B10_SINT_PACK32: return getTranslatedFinalFormat(FORMAT_RGB10A2_SINT_PACK32);	//GL_RGB10_A2I
+
+			// signed integer formats
+			case EF_R8_SINT: return getTranslatedFinalFormat(FORMAT_R8_SINT_PACK8);				//GL_R8I
+			case EF_R8G8_SINT: return getTranslatedFinalFormat(FORMAT_RG8_SINT_PACK8);				//GL_RG8I
+			case EF_R8G8B8_SINT: return getTranslatedFinalFormat(FORMAT_RGB8_SINT_PACK8);			//GL_RGB8I
+			case EF_R8G8B8A8_SINT: return getTranslatedFinalFormat(FORMAT_RGBA8_SINT_PACK8);			//GL_RGBA8I
+
+			case EF_R16_SINT: return getTranslatedFinalFormat(FORMAT_R16_SINT_PACK16);				//GL_R16I
+			case EF_R16G16_SINT: return getTranslatedFinalFormat(FORMAT_RG16_SINT_PACK16);			//GL_RG16I
+			case EF_R16G16B16_SINT: return getTranslatedFinalFormat(FORMAT_RGB16_SINT_PACK16);			//GL_RGB16I
+			case EF_R16G16B16A16_SINT: return getTranslatedFinalFormat(FORMAT_RGBA16_SINT_PACK16);			//GL_RGBA16I
+
+			case EF_R32_SINT: return getTranslatedFinalFormat(FORMAT_R32_SINT_PACK32);				//GL_R32I
+			case EF_R32G32_SINT: return getTranslatedFinalFormat(FORMAT_RG32_SINT_PACK32);			//GL_RG32I
+			case EF_R32G32B32_SINT: return getTranslatedFinalFormat(FORMAT_RGB32_SINT_PACK32);			//GL_RGB32I
```

Comment:
formats which don't fit in Power-Of-Two for a texel/block will never be allowed to be texture formats (only vertex formats)